lib/APPTurtle/src/App.cpp

```diff
@@ -295,57 +294,38 @@ void App::handleTurtle()
     /* Check for new data. */
     if (true == m_isNewTurtleSpeedSetpoint)
     {
-        /*
-        Max speed is currently unknown to DCS, but using a big value will ensure that
-        RU drives as fast as possible as it will cap the speed to its own maximum.
-        */
-        const int16_t MOTOR_MAX_SPEED = 5000;
-        SpeedData     payload;
-
-        /*
-        Move turtle regardless of the exact speed setpoint received.
-        */
-        if (m_turtleSpeedSetpoint.linear.x > 0.0F)
-        {
-            payload.left  = MOTOR_MAX_SPEED;
-            payload.right = MOTOR_MAX_SPEED;
-        }
-        /* Turn left. */
-        else if (m_turtleSpeedSetpoint.angular.z > 0.0F)
-        {
-            payload.left  = -MOTOR_MAX_SPEED;
-            payload.right = MOTOR_MAX_SPEED;
-        }
-        /* Turn right. */
-        else if (m_turtleSpeedSetpoint.angular.z < 0.0F)
-        {
-            payload.left  = MOTOR_MAX_SPEED;
-            payload.right = -MOTOR_MAX_SPEED;
-        }
+        TurtleSpeed payload;
+        int32_t     linearSpeed  = m_turtleSpeedSetpoint.linear.x * 1000;  /* Linear speed in mm/s */
```

Comment:
Scale is a magic number, better use a constant or inline to make it clear. From code it is not obvious what setpoint.linear is.

Example:
/**
   * Scale linear speed from twist message (meter/s) to milimeter/s
   */
inline  int32_t scaleTwistSpeed(int32_t speed)
{
   return meterPerSec *1000;  /* to mm/s *
}